test/functional/wallet_conflicts.py

```diff
@@ -0,0 +1,113 @@
+
+        outputs_conflicted_tx = {}
+        outputs_conflicted_tx[self.nodes[0].getnewaddress()] = Decimal("19.99998")
+        outputs_conflicting_tx_A = {}
+        outputs_conflicting_tx_A[self.nodes[0].getnewaddress()] = Decimal("9.99998")
+        outputs_conflicting_tx_B = {}
+        outputs_conflicting_tx_B[self.nodes[0].getnewaddress()] = Decimal("9.99998")
+
+        conflicted = self.nodes[0].signrawtransactionwithwallet(self.nodes[0].createrawtransaction(inputs_conflicted_tx, outputs_conflicted_tx))
+        conflicting_tx_A = self.nodes[0].signrawtransactionwithwallet(self.nodes[0].createrawtransaction(inputs_conflicting_tx_A, outputs_conflicting_tx_A))
+        conflicting_tx_B = self.nodes[0].signrawtransactionwithwallet(self.nodes[0].createrawtransaction(inputs_conflicting_tx_B, outputs_conflicting_tx_B))
+
+        self.log.info("Broadcast conflicted transaction")
+        conflicted_txid = self.nodes[0].sendrawtransaction(conflicted["hex"])
+        self.generate(self.nodes[0], 1, sync_fun=self.no_op)
+
+        # Build child conflicted tx
+        nA = next(tx_out["vout"] for tx_out in self.nodes[0].gettransaction(conflicted_txid)["details"] if tx_out["amount"] == Decimal("19.99998"))
+        inputs_child_conflicted = []
+        inputs_child_conflicted.append({"txid": conflicted_txid, "vout": nA})
+        outputs_child_conflicted = {}
+        outputs_child_conflicted[self.nodes[0].getnewaddress()] = Decimal("19.99996")
+        child_conflicted = self.nodes[0].signrawtransactionwithwallet(self.nodes[0].createrawtransaction(inputs_child_conflicted, outputs_child_conflicted))
+        child_conflicted_txid = self.nodes[0].sendrawtransaction(child_conflicted["hex"])
+        self.generate(self.nodes[0], 1, sync_fun=self.no_op)
+
+        self.log.info("Broadcast conflicting tx to node 1")
+        conflicting_txid_A = self.nodes[1].sendrawtransaction(conflicting_tx_A["hex"])
+        self.generate(self.nodes[1], 4, sync_fun=self.no_op)
+        self.nodes[1].sendrawtransaction(conflicting_tx_B["hex"])
+        self.generate(self.nodes[1], 4, sync_fun=self.no_op)
+
+        # Reconnect node0 and node1 and check that conflicted_txid is effectively conflicted
+        self.log.info("Connect nodes 0 and 1, and ensure that the tx is effectively conflicted")
+        self.connect_nodes(0, 1)
+        self.sync_blocks([self.nodes[0], self.nodes[1]])
+        conflicted = self.nodes[0].gettransaction(conflicted_txid)
+        child_conflicted = self.nodes[0].gettransaction(child_conflicted_txid)
+        conflicting = self.nodes[0].gettransaction(conflicting_txid_A)
+        # Conflicted tx should have confirmations set to the confirmations of the most conflicting tx
+        assert_equal(conflicted["confirmations"], -conflicting["confirmations"])
+        # Child should inherit conflicted state from parent
+        assert_equal(child_conflicted["confirmations"], -conflicting["confirmations"])
```

Comment:
We should also test who "wins" in this conflict contest and what the values are, i.e. conflicting A has 8 confirmations and conflicting B has 4 confirmations.

Reply:
I've added a check for this.